src/components/LoansYouMightLike/lymlContainer.vue

```diff
@@ -192,15 +192,17 @@ export default {
 				loansYouMightLike.push(randomLoans[0]);
 
 				// same Country loans
-				if (this.hasLoansInBasket && data.data.lend.sameCountry) {
-					loansYouMightLike.push(_get(data.data.lend, 'sameCountry.values[1]'));
+				const sameCountryLoans = _get(data, 'data.lend.sameCountry') || [];
+				if (sameCountryLoans.length > 1) {
+					loansYouMightLike.push(sameCountryLoans.values[1]);
 				} else {
 					loansYouMightLike.push(randomLoans[1]);
 				}
 
 				// same Activity loans
-				if (this.hasLoansInBasket && data.data.lend.sameActivity) {
-					loansYouMightLike.push(_get(data.data.lend, 'sameActivity.values[1]'));
+				const sameActivityLoans = _get(data, 'data.lend.sameActivity') || [];
+				if (sameActivityLoans.length > 1) {
+					loansYouMightLike.push(sameActivityLoans.values[1]);
 				} else {
 					loansYouMightLike.push(randomLoans[2]);
 				}
```

Comment:
These if/else statements checking for the data and if it's not there passing through random loans make sense for sure. Thanks for tagging me here on the solution. 👍